File: Vjudge/xeta_long-1/A-Where is the marvel.cpp

#include<bits/stdc++.h>
using namespace std;

int main()
{
    int N,Q,cas=0;
    while(cin>>N>>Q)
    {
        if(N==0 &&Q==0){
            break;
        }
        printf("CASE# %d:\n", ++cas);
        int str[10005];
        for (int i = 0; i<N; i++)
        {
            cin>>str[i];
        }
        sort(str, str+N);
        while(Q--){
            int que, answer;
            cin>>que;
            answer = lower_bound(str,str+N, que)-str;
            if (str[answer]==que){
               cout<<que<<" found at "<<answer+1<<endl;

            }
            else{
                cout<<que<<" not found"<<endl;
            }
        }
    }
}

File: Vjudge/xeta_long-1/A-Where is the marvel.py
from bisect import bisect_left, bisect_right

while True:
    case = 0
    N, Q = map(int, input().split(" "))
    if N == 0 and Q == 0:
        break
    print("CASE# ", ++case)
    lst = []
    for i in range(N):
        a = int(input())
        lst.append(a)
    lst.sort()
    # print(lst)
    while Q==False:
        que = int(input())
        answer = bisect_left(a, lst, que)
        if lst[answer] == que:
            print(que, "fround at", answer+1)
        else:
            print("not found")


